apps/client/src/common/hooks/useFollowComponent.ts

```diff
@@ -0,0 +1,61 @@
+import { MutableRefObject, useCallback, useEffect } from 'react';
+
+function scrollToComponent<ComponentRef extends HTMLElement, ScrollRef extends HTMLElement>(
+  componentRef: MutableRefObject<ComponentRef>,
+  scrollRef: MutableRefObject<ScrollRef>,
+  topOffset: number,
+) {
+  if (!componentRef.current || !scrollRef.current) {
+    return;
+  }
+
+  const componentRect = componentRef.current.getBoundingClientRect();
+  const scrollRect = scrollRef.current.getBoundingClientRect();
+  const top = componentRect.top - scrollRect.top + scrollRef.current.scrollTop - topOffset;
+
+  scrollRef.current.scrollTo({ top, behavior: 'smooth' });
+}
+
+interface UseFollowComponentProps {
+  followRef: MutableRefObject<HTMLElement | null>;
+  scrollRef: MutableRefObject<HTMLElement | null>;
+  doFollow: boolean;
+  topOffset?: number;
+}
+
+export default function useFollowComponent(props: UseFollowComponentProps) {
+  const { followRef, scrollRef, doFollow, topOffset = 100 } = props;
```

Comment:
hmm, I never thought about that.
I guess the prop terminology is very react like. The conventional terminology for a function would be arguments or params

This is a good observation, the short answer is that it might not matter other than for semantics. In the current code base props has been used so I will stick to that. For my next project I will keep that in mind and see what happens

Out of curiosity, just snooped around on sentry
https://github.com/getsentry

and it seems they tend to use the term Option, so `useFollowComponentOptions`, which makes sense